#7   Added the score feature correctly

GhostUser opened the pull request:
Thank you for the correct procedure. I'm still a beginner in version control so that happened by mistake. From next time I'll take care that these sort of issues don't take place. 

Once again sorry again for causing any kind of trouble.

Thank You

GhostUser committed:
Added the score feature correctly

douglascdev commented on snake/main.py:
```diff
@@ -7,6 +7,8 @@
 from pygame import event, gfxdraw
 
 
+pygame.init()
```
This initializes all the pygame modules, but we only need font and display for now, so you can initialize only the font. The place you put this line is also not right, you should put it where I'm already initializing the display, to keep them in the same place.

douglascdev commented on snake/main.py:
```diff
@@ -21,7 +23,7 @@ class Game:
     NUM_UNITS_RESPAWN = 3
     FRAMETIME_WAITED = 64
-
+    score=0
```
According to [PEP8](https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements), you should keep a single whitespace on each side of the assignment operator, as I also said on your first pull request.
 I'd suggest you move this line to the __main__ part of the code as well, since I meant for these classes to hold [constants](https://www.python.org/dev/peps/pep-0008/#constants)(I'll put them in a separate file later to keep that more clear).

douglascdev commented on snake/main.py:
```diff
@@ -30,6 +32,7 @@ class Screen:
     WIDTH, HEIGHT = block_size * NUM_BLOCKS_X, block_size * NUM_BLOCKS_Y
     RECT = Rect(0, 0, WIDTH, HEIGHT)
     FPS = 60
+    
```
Adding a line here is also against PEP8, as you can see [here](https://www.python.org/dev/peps/pep-0008/#blank-lines).

douglascdev commented on snake/main.py:
```diff
@@ -102,6 +105,7 @@ def update(self, frametime: int, food_group: pygame.sprite.Group):
                 y_mov = self.direction.value[1]
                 new_head.rect = head.rect.move(x_mov, y_mov)
                 self.add(new_head)
+                Game.score+=1
```
Another whitespace missing here

douglascdev commented on snake/main.py:
```diff
@@ -189,6 +193,11 @@ def checkered_surface(screen: pygame.Surface) -> pygame.Surface:
+def score_update(score):
+    font = pygame.font.SysFont("sans-sarif", 35, True)
+    text = font.render("Score: " + str(score), 100, Color.WHITE) # Arguments are: text, anti-aliasing, color
+    pg_screen.blit(text, (650, 10))
```
That blit line has a couple of issues. 
 The first is that you're referencing pg_screen, a variable that is from outer scope. It works now because pg_screen is defined when your function gets called, but you don't know if your code will still work in the future, if someone makes changes to pg_screen in the main code and then call your function. To keep that from happening, you could pass the screen surface to your function as a parameter, instead.
 The second is that you hard coded the position where the font will be drawn. A change in resolution or font size would break your code, making the score appear in the wrong position(you can increase DEFAULT_RECT_SIZE to like 60 if you want to see this by yourself). To keep it in the upper right corner, you need to calculate the text's position according to the text and screen sizes. Let me know if you need help with that.

douglascdev reviewed (changes requested):
No problem, thank you for contributing and taking your time to make changes. Just push these changes to your branch and I'll merge.

douglascdev commented on snake/main.py:
```diff
@@ -189,6 +193,11 @@ def checkered_surface(screen: pygame.Surface) -> pygame.Surface:
     return checkered
 
+def score_update(score):
+    font = pygame.font.SysFont("sans-sarif", 35, True)
```
I don't think sans-sarif is a real font lol
Try pygame.font.get_fonts() to see the available fonts in your system

GhostUser committed:
Did the recquired changes

GhostUser commented:
Changed the font and removed the constants in blit. Now, the score can change size according to DEFAULT_RECT_SIZE.  Removed score from Game class and added  it to main.

GhostUser committed:
.

douglascdev committed:
Reformat code using black

douglas-cpp commented:
You didn't fix the style, so I made another commit on your branch
